src/productcatalogservice/Dockerfile

```diff
@@ -11,16 +11,27 @@ COPY Gopkg.* ./
 RUN dep ensure --vendor-only -v
 
 COPY . .
-RUN go build -o /productcatalogservice .
+RUN go build -gcflags=all='-N -l' -o /productcatalogservice .
 
 FROM alpine AS release
-RUN apk add --no-cache ca-certificates
+RUN apk add --no-cache ca-certificates \
+            libc6-compat
+
 RUN GRPC_HEALTH_PROBE_VERSION=v0.2.0 && \
     wget -qO/bin/grpc_health_probe https://github.com/grpc-ecosystem/grpc-health-probe/releases/download/${GRPC_HEALTH_PROBE_VERSION}/grpc_health_probe-linux-amd64 && \
     chmod +x /bin/grpc_health_probe
+
 WORKDIR /productcatalogservice
+
+RUN wget -O go-cloud-debug https://storage.googleapis.com/cloud-debugger/compute-go/go-cloud-debug \
+            && chmod 0755 go-cloud-debug
+
 COPY --from=builder /productcatalogservice ./server
 COPY products.json .
+COPY source-context.json .
+
 EXPOSE 3550
-ENTRYPOINT ["/productcatalogservice/server"]
+ENTRYPOINT ["./go-cloud-debug", "-sourcecontext=source-context.json", \
```

Comment:
Is it possible to not to change the entrypoint?

I don't think people actually deploy applications with a debugger wrapper. One of the goals of the sample app is to show best practices, so if there's a way to avoid this, it would be great.

Reply:
Unfortunately, that's the official way of using the golang stackdriver debugger: https://cloud.google.com/debugger/docs/setup/go#gce